src/main/java/uk/gov/hmcts/reform/civil/helpers/BundleRequestMapper.java

```diff
@@ -185,88 +186,101 @@ private List<Element<BundlingRequestDocument>> mapSystemGeneratedcaseDocument(Li
-                                           .build());
-
-            });
+        if (null == uploadEvidenceExpert || uploadEvidenceExpert.isEmpty()) {
+            return ElementUtils.wrapElements(bundlingExpertDocs);
         }
+
+        uploadEvidenceExpert.forEach(expertDocs -> {
+            StringBuilder fileNameBuilder = new StringBuilder();
+            fileNameBuilder.append(displayName);
+            if (Optional.ofNullable(expertDocs.getValue().getExpertOptionName()).isPresent()) {
+                fileNameBuilder.append("_" + expertDocs.getValue().getExpertOptionName());
+            }
+            if (Optional.ofNullable(expertDocs.getValue().getExpertOptionUploadDate()).isPresent()) {
+                fileNameBuilder.append("_" + DateFormatHelper.formatLocalDate(
+                    expertDocs.getValue()
+                        .getExpertOptionUploadDate(),
+                    "ddMMyyyy"
+                ));
+            }
+            Document document = expertDocs.getValue().getExpertDocument();
+            bundlingExpertDocs.add(BundlingRequestDocument.builder()
+                                       .documentFileName(fileNameBuilder.toString())
+                                       .documentLink(DocumentLink.builder()
+                                                         .documentUrl(document.getDocumentUrl())
+                                                         .documentBinaryUrl(document.getDocumentBinaryUrl())
+                                                         .documentFilename(document.getDocumentFileName()).build())
+                                       .build());
+
+        });
+
         return ElementUtils.wrapElements(bundlingExpertDocs);
     }
 
     private List<Element<BundlingRequestDocument>> mapUploadEvidenceOtherDoc(List<Element<UploadEvidenceDocumentType>> otherDocsEvidenceUpload) {
         List<BundlingRequestDocument> bundlingExpertDocs = new ArrayList<>();
-        if (!Optional.ofNullable(otherDocsEvidenceUpload).isEmpty()) {
-            otherDocsEvidenceUpload.forEach(otherDocs -> {
-                Document document = otherDocs.getValue().getDocumentUpload();
-                bundlingExpertDocs.add(BundlingRequestDocument.builder()
-                                           .documentFileName(document.getDocumentFileName())
-                                           .documentLink(DocumentLink.builder()
-                                                             .documentUrl(document.getDocumentUrl())
-                                                             .documentBinaryUrl(document.getDocumentBinaryUrl())
-                                                             .documentFilename(document.getDocumentFileName()).build())
-                                           .build());
-
-            });
+        if (null == otherDocsEvidenceUpload || otherDocsEvidenceUpload.isEmpty()) {
```

Comment:
See above comments. Consider the following:

```suggestion
        if (null == otherDocsEvidenceUpload) {

```